### PR TITLE
SocketAsyncEngine.Unix: name socket event thread

### PR DESCRIPTION
**Pull request by tmds**
This makes it more easy to identify the thread amongst
the other threads in the application.

cc @stephentoub @dotnet/ncl

### CODE REVIEW

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs [hunk 1]
```diff
@@ -277,24 +277,9 @@ private SocketAsyncEngine()
...
-                        CancellationToken.None,
-                        TaskCreationOptions.LongRunning,
-                        TaskScheduler.Default);
-                }
-                finally
-                {
-                    if (suppressFlow) ExecutionContext.RestoreFlow();
-                }
+                Thread thread = new Thread(s => ((SocketAsyncEngine)s!).EventLoop());
+                thread.Name = ".NET Sockets";
```
Comment (tmds) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:281]: Other threads use `.NET` as prefix, e.g. `.NET Finalizer` so I've done that too.
Other name: `.NET Events`?
`.NET SocketEvents` is too long, it gets truncated to `.NET SocketEven`.

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs [hunk 2]
```diff
@@ -277,24 +277,9 @@ private SocketAsyncEngine()
                     throw new InternalException(err);
                 }
 
-                //
-                // Start the event loop on its own thread.
-                //
-                bool suppressFlow = !ExecutionContext.IsFlowSuppressed();
-                try
-                {
-                    if (suppressFlow) ExecutionContext.SuppressFlow();
```
Comment (stephentoub) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:286]: Why did you remove the execution context stuff? That applies to Thread equally.

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs [hunk 3]
```diff
@@ -277,24 +277,9 @@ private SocketAsyncEngine()
...
-                        this,
-                        CancellationToken.None,
-                        TaskCreationOptions.LongRunning,
-                        TaskScheduler.Default);
-                }
-                finally
-                {
-                    if (suppressFlow) ExecutionContext.RestoreFlow();
-                }
+                Thread thread = new Thread(s => ((SocketAsyncEngine)s!).EventLoop());
```
Comment (stephentoub) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:280]: The thread needs to have IsBackground set to true.

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:281]: The name you choose seems fine.

**Review by stephentoub — COMMENTED** [review]
Thanks

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs (hunk 2 above)
Comment (tmds) [reply src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:286]: I assumed otherwise. Adding this back.

**Review by stephentoub — APPROVED** [review]

**Comment by danmosemsft** [comment]
XUnit exception:
```
System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument) in /_/src/libraries/System.Linq/src/System/Linq/ThrowHelper.cs:line 13
   at System.Linq.Enumerable.Cast[ITestCase](IEnumerable source) in /_/src/libraries/System.Linq/src/System/Linq/Cast.cs:line 42
   at Xunit.Sdk.TestMethodRunner`1.<RunAsync>d__31[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestMethodRunner.cs:line 117
   at Xunit.Sdk.TestClassRunner`1.<RunTestMethodsAsync>d__38[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestClassRunner.cs:line 213
   at Xunit.Sdk.TestClassRunner`1.<RunAsync>d__37[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestClassRunner.cs:line 171
   at Xunit.Sdk.TestCollectionRunner`1.<RunTestClassesAsync>d__28[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestCollectionRunner.cs:line 130
   at Xunit.Sdk.TestCollectionRunner`1.<RunAsync>d__27[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestCollectionRunner.cs:line 101
   at Xunit.Sdk.XunitTestAssemblyRunner.RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\XunitTestAssemblyRunner.cs:line 205
   at Xunit.Sdk.TestAssemblyRunner`1.<RunAsync>d__41[[Xunit.Sdk.IXunitTestCase, xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]].MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestAssemblyRunner.cs:line 196
   at Xunit.Sdk.XunitTestFrameworkExecutor.RunTestCases(IEnumerable`1 testCases, IMessageSink executionMessageSink, ITestFrameworkExecutionOptions executionOptions) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\XunitTestFrameworkExecutor.cs:line 95
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:line 1892
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs:line 871
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs:line 664
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 38
   at System.Threading.Thread.StartCallback() in /_/src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs:line 279
```

**Comment by tmds** [comment]
> XUnit exception:

This CI fail is unrelated.

@stephentoub thank you for reviewing!
This should be good to merge.
